devices/src/virtio/net.rs

```diff
@@ -511,22 +511,26 @@ impl EpollHandler for NetEpollHandler {
         match device_event {
             RX_QUEUE_EVENT => {
                 METRICS.net.rx_queue_event_count.inc();
-                if let Err(e) = self.rx.queue_evt.read() {
-                    error!("Failed to get rx queue event: {:?}", e);
-                    METRICS.net.event_fails.inc();
-                    Err(DeviceError::FailedReadingQueue {
-                        event_type: "rx queue event",
-                        underlying: e,
+                self.rx
```

Comment:
nit: old-timers like myself like the old-fashioned `if/else` constructs. Makes the branching more visible as opposed to one long line of methods and closures. Especially in this case where, for example, the `and_then(|_| {})` closure is not even using its params.

This is just personal preference and will let others chime in as well.

Reply:
Btw, I think you can omit the closure parameters entirely when no param is required, so you could write `and_then(|| {})` instead of `and_then (|_|, {})`. What I don't like about `and_then` in this particular case is the extra indentation :(

Reply:
> Btw, I think you can omit the closure parameters entirely when no param is required, so you could write `and_then(|| {})` instead of `and_then (|_|, {})`. What I don't like about `and_then` in this particular case is the extra indentation :(

Hmm, I think it might be for something else, and if the method requires a 1 param lambda, you have to pass it a 1 param lambda, otherwise

```
--> devices/src/virtio/net.rs:525:22
    |
525 |                     .and_then(|| {
    |                      ^^^^^^^^ -- takes 0 arguments
    |                      |
    |                      expected closure that takes 1 argument
help: consider changing the closure to take and ignore the expected argument
    |
525 |                     .and_then(|_| {
    |                               ^^^
```

Reply:
> nit: old-timers like myself like the old-fashioned `if/else` constructs. Makes the branching more visible as opposed to one long line of methods and closures. Especially in this case where, for example, the `and_then(|_| {})` closure is not even using its params.
> 
> This is just personal preference and will let others chime in as well.

I'm fine either way, but @aghecenco mentioned it in https://github.com/firecracker-microvm/firecracker/pull/793#discussion_r244319517 and it made sense to me :)

Reply:
Understood; I've pushed a revert of 97e9ffd0c862ceb0d79393b11b1a4a329e57eac3